src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -719,6 +719,7 @@ internal bool ShouldUpdateChild(TrieNode? parent, TrieNode? oldChild, TrieNode?
             if (oldChild is null && newChild is null) return false;
             if (!ReferenceEquals(oldChild, newChild)) return true;
             if (newChild.Keccak is null && parent.Keccak is not null) return true; // So that recalculate root knows to recalculate the parent root.
+            if (newChild.Keccak is null && parent.Keccak is null) return true; // parent's hash can also be null depending on nesting level - still need to update child, otherwise combine will remain original value
```

Comment:
If only bulkset is failing, can you make another `ShouldUpdateChild` but just for bulk? This method is used by setting one by one too.

Reply:
Yes, so the case is gradually traversing down until one by one, so I think it makes sense for it to remain global